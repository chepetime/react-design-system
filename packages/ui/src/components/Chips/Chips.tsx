import React from "react";
import classnames from "classnames";
import css from "./Chips.module.scss";

export interface ChipsProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Chips = React.forwardRef<HTMLDivElement, ChipsProps>(
  (props: ChipsProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Chips)} ref={ref}>
        <p>Chips component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Chips.displayName = "Chips";
