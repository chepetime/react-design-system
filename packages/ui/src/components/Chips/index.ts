export { Chips } from "./Chips";
export type { ChipsProps } from "./Chips";