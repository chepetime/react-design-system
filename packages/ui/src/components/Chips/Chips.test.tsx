
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Chips } from "./Chips";
import { render, screen } from "@testing-library/react";

describe("Chips", () => {
  it("renders without errors", () => {
    render(<Chips />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Chips>Some children</Chips>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Chips>Some children</Chips>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Chips />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
