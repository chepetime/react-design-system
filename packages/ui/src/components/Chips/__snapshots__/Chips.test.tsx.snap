// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips matches snapshot 1`] = `
<div>
  <div
    class="Chips"
  >
    <p>
      Chips component working!
    </p>
    <div />
  </div>
</div>
`;
